app/deploy/support/amq63.yml

```diff
@@ -0,0 +1,138 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: amq63
+message: A new messaging service has been created in your project. It will handle
+  the protocol openwire. The username/password for accessing the service
+  is ${MQ_USERNAME}/${MQ_PASSWORD}.
+metadata:
+  annotations:
+    description: Application template for JBoss A-MQ brokers. These can be deployed
+      as standalone or in a mesh. This template doesn't feature SSL support.
+    iconClass: icon-amq
+    openshift.io/display-name: Red Hat JBoss A-MQ 6.3 (Ephemeral, no SSL)
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: messaging,amq,jboss
+  name: amq63-basic
+  parameters:
+  - description: The name for the application.
+    displayName: Application Name
+    name: APPLICATION_NAME
+    required: true
+    value: broker
+  - description: Queue names, separated by commas. These queues will be automatically
+      created when the broker starts. If left empty, queues will be still created dynamically.
+    displayName: Queues
+    name: MQ_QUEUES
+  - description: Topic names, separated by commas. These topics will be automatically
+      created when the broker starts. If left empty, topics will be still created dynamically.
+    displayName: Topics
+    name: MQ_TOPICS
+  - description: User name for standard broker user. It is required for connecting to
+      the broker. If left empty, it will be generated.
+    displayName: A-MQ Username
+    value: amq
+    name: MQ_USERNAME
+  - description: Password for standard broker user. It is required for connecting to
+      the broker. If left empty, it will be generated.
+    displayName: A-MQ Password
+    value: topSecret
```

Comment:
Setting a value doesn't match with the description: `If left empty, it will be generated.`

I'm not keen on hard coding passwords like this, but if it's the same as for the DB connector, an example one, then i'm happy enough with it I guess.

Reply:
I'll bring back the generators. It seems smoother and can be easily overwritten by user.